### ISSUE TITLE
when use untrainded data to predict, the output is so bad.

### ISSUE THREAD

**Issue by xinsuinizhuan** [josephjaspers/blackcat_tensors#60]
I trained my data, epoch == 3000, and when i use the trained data to predict, the result is so good, but when i use the closed data to predict, the output is so bad. I don't know whether this is right or wrong? But the normal mode to evaluate it, the data need to divide into trained data and test data, traiend use the trained data and evaluate it by test data.

**Comment by josephjaspers** [comment]
Sounds like the neural-network is overfitting the trained data and not learning to generalize properly. 
This is a fairly common.

You most likely do not need 3000 epochs in your case, try decreasing and seeing if test data accuracy improves

**Comment by xinsuinizhuan** [comment]
OK.Let me try!

**Comment by xinsuinizhuan** [comment]
I have try it. when i set epoch == 1000, use the untrained data predict, the result is also not good, so bad.
